### PR TITLE
PYTHON-1853 Empty projections should return the entire document not just the _id

### PR DESCRIPTION
**Pull request by juliusgeo**

### CODE REVIEW

File: test/test_collection.py [hunk 1]
```diff
@@ -1707,7 +1707,8 @@ def test_find_one(self):
         self.assertTrue("hello" in db.test.find_one(projection=frozenset(["hello"])))
         self.assertTrue("hello" not in db.test.find_one(projection=frozenset(["foo"])))
 
-        self.assertEqual(["_id"], list(db.test.find_one(projection=[])))
+        self.assertEqual(["_id"], list(db.test.find_one(projection={'_id':
+            True})))
```
Comment (ShaneHarvey) [inline test/test_collection.py:1711]: Please add a new test for the explicitly empty projection case (ie projection={} and projection=[]).

File: pymongo/cursor.py [hunk 2]
```diff
@@ -194,9 +194,7 @@ def __init__(self, collection, filter=None, projection=None, skip=0,
         if allow_disk_use is not None:
             allow_disk_use = validate_boolean("allow_disk_use", allow_disk_use)
 
-        if projection is not None:
-            if not projection:
-                projection = {"_id": 1}
+        if projection not in (None, {}):
```
Comment (ShaneHarvey) [inline pymongo/cursor.py:197]: We should remove the {} from this check and go back to `is not None:`. Checking for `{}` is unnecessary.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Please remember to update the changelog, migration guides, and method docstring for collection.find to mention this breaking change.

File: pymongo/cursor.py (hunk 2 above)
Comment (juliusgeo) [reply pymongo/cursor.py]: Done.

**Comment by behackett** [comment]
> Please remember to update the changelog, migration guides, and method docstring for collection.find to mention this breaking change.

Agreed. The existing behavior has existed since PyMongo 0.14. That's a long time. We need to carefully document this behavior change and how to get back the previous behavior for users that have come to rely on it.

File: test/test_collection.py (hunk 1 above)
Comment (juliusgeo) [reply test/test_collection.py:1711]: Done.

Comment (ShaneHarvey) [reply test/test_collection.py:1711]: Please add a test for `[]` as well.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: doc/migrate-to-pymongo4.rst [hunk 3]
```diff
@@ -637,6 +637,20 @@ Can be changed to this::
 
 You must now explicitly compare with None.
 
+Collection.find returns entire document with empty projection
+.............................................................
+Empty (None, {})  projections for find commands are passed on to the server
```
[the remark this answers is not in the document]
Comment (juliusgeo) [reply doc/migrate-to-pymongo4.rst]: Done.

File: test/test_collection.py (hunk 1 above)
Comment (juliusgeo) [reply test/test_collection.py:1711]: Done.

File: pymongo/collection.py [hunk 4]
```diff
@@ -1290,6 +1290,16 @@ def find(self, *args, **kwargs):
            expression object.
            Soft deprecated the ``manipulate`` option.
 
+        .. versionchanged:: 4.0
```
Comment (ShaneHarvey) [inline pymongo/collection.py:1293]: Please merge this section with the existing `versionchanged:: 4.0` above.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/collection.py (hunk 4 above)
Comment (juliusgeo) [reply pymongo/collection.py]: Done.

File: pymongo/collection.py [hunk 5]
```diff
@@ -1246,6 +1246,13 @@ def find(self, *args, **kwargs):
 
         .. versionchanged:: 4.0
            Removed the ``modifiers`` option.
+           Empty projections (eg {} or []) for
+           :meth:`~pymongo.collection.Collection.find`, and
+           :meth:`~pymongo.collection.Collection.find_one`
```
Comment (ShaneHarvey) [inline pymongo/collection.py:1251]: I think we can remove the "for find and find_one" part here because this is the docs for find.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]

File: pymongo/collection.py (hunk 5 above)
Comment (juliusgeo) [reply pymongo/collection.py]: Makes sense. Fixed.

**Review by ShaneHarvey — APPROVED** [review]
